app/models/Property.test.ts

```diff
@@ -1,56 +1,183 @@
-});
-
-it("correctly calculates the landPrice", () => {
-  expect(property.landPrice).toBeCloseTo(31531.579);
-});
-
-it("correctly calculates the landToTotalRatio", () => {
-  expect(property.landToTotalRatio).toBeCloseTo(0.14);
-});
-
-it("correctly calculates the values even for number of bedroooms exceeding the max ", () => {
-  property = new Property({
-    postcode: "WV8 1HG",
-    houseType: "T",
-    numberOfBedrooms: 20,
-    age: 9,
-    size: 88,
-    maintenancePercentage: 0.02,
-    newBuildPricePerMetre: 2120,
-    averageMarketPrice: 218091.58,
-    itl3: "TLG24",
-  });
-  expect(property).toBeInstanceOf(Property);
-});
+describe('Property', () => {
+  beforeEach(() => {
+    property = new Property({
+      postcode: "MK14 7AG",
+      houseType: "T",
+      numberOfBedrooms: 2,
+      age: 10,
+      size: 88,
+      maintenancePercentage: 0.02,
+      newBuildPricePerMetre: 2120,
+      averageMarketPrice: 219135,
+      itl3: "TLJ12",
+    });
+  });
+
+  let property: Property;
+
+  it("can be instantiated", () => {
+    expect(property).toBeInstanceOf(Property);
+  });
+  
+  it("correctly calculates the newBuildPrice", () => {
+    expect(property.newBuildPrice).toBeCloseTo(186560);
+  });
+
+  // 
+  it("correctly calculates the bedWeightedAveragePrice", () => {
+    expect(property.bedWeightedAveragePrice).toBeCloseTo(219135);
+  });
+  
+  it("correctly calculates the landPrice", () => {
+    expect(property.landPrice).toBeCloseTo(32575);
+  });
+  
+  it("correctly calculates the landToTotalRatio", () => {
+    expect(property.landToTotalRatio).toBeCloseTo(0.1446);
+  });
+  
+  it("correctly calculates the values even for number of bedroooms exceeding the max ", () => {
+    property = new Property({
+      postcode: "WV8 1HG",
+      houseType: "T",
+      numberOfBedrooms: 20,
+      age: 11,
+      size: 88,
+      maintenancePercentage: 0.02,
+      newBuildPricePerMetre: 2120,
+      averageMarketPrice: 218091.58,
+      itl3: "TLG24",
+    });
+    expect(property).toBeInstanceOf(Property);
+  });
+
+  it("correctly returns newBuildPrice if newbuild", () => {
+    property = new Property({
+      postcode: "WV8 1HG",
+      houseType: "T",
+      numberOfBedrooms: 20,
+      age: 1,
+      size: 88,
+      maintenancePercentage: 0.02,
+      newBuildPricePerMetre: 2120,
+      averageMarketPrice: 218091.58,
+      itl3: "TLG24",
+    });
+
+    expect(property.depreciatedBuildPrice).toBe(property.newBuildPrice);
+  });
+
+  describe('depreciation calculations (existing build)', () => {
+
+    it("correctly calculates newComponentValue for foundations", () => {
+      const result = property.calculateComponentValue(
+        'foundations',
+        property.newBuildPrice,
+        property.age,
+        MAINTENANCE_LEVELS[0]
+      );
+
+      const expectedNewComponentValue = 
+        property.newBuildPrice * 
+        HOUSE_BREAKDOWN_PERCENTAGES.foundations.percentageOfHouse;
```

Comment:
I'm personally don't think that this approach of repeating our logic within our tests is the right one. This might be one of the rare places where a magic number is correct.

This approach - 
 - Is more complex to write
 - Is more complex to maintain (any changes need to be updated in many places)
 - Allows us to make a mistake in logic in the tests and class, but having a passing unit test
 
 This sort of sums it up I think - https://aaronsaray.com/2021/do-not-calculate-answers-to-tests/

Reply:
Cheers for pointing this out! 
> “I’m using different mechanisms to calculate the response values, so the chance both of those are wrong is very small.” You’re right, except you’re not.

😅

The argument here makes sense to me, I'm just struggling with how to come up with that "known answer" in the first place. If the number is complicated to arrive at, how else will you arrive at it if not through code? I don't actually _know_ what number I expect to come out of the code 🤔 I'll have a go tomorrow at refactoring all of the tests and trying to come up with another way of generating the expected values.